opencodelists/tests/functional/test_docs_screenshots.py

```diff
@@ -140,3 +205,226 @@ def test_generate_org_user_screen_shots(
         "my-organisations.png",
         extra={"height": -2},
     )
+
+    # Create a codelist button.
+    page.get_by_role("link", name="My codelists").click()
+    screenshot_element_with_padding(
+        page,
+        page.get_by_role("link", name="Create a codelist!"),
+        "create-codelist-button.png",
+    )
+
+    page.get_by_role("link", name="Create a codelist!").click()
+
+    page.get_by_role("textbox", name="Codelist name *").fill("Tennis elbow")
+    page.get_by_label("Coding system *").select_option("snomedct")
+
+    # Codelist creation page.
+    screenshot_element_with_padding(
+        page,
+        page.locator("div.container > form"),
+        "create-new-codelist-page.png",
+        extra={"y": 90, "height": -170},
+        full_page=True,
+    )
+
+    page.get_by_role("button", name="Create").click()
+
+    # Initial codelist builder.
+    # Strictly, we should already be able to take a screenshot of the builder.
+    # However, without clicking somewhere first, there is seemingly
+    # inconsistent behaviour where the height of the image
+    # gets truncated some of the time.
+    # This is despite using a builder-specific locator.
+    # Clicking somewhere after page load seems to make it reliable.
+    page.get_by_role("heading", name="Create a codelist").click()
+
+    screenshot_element_with_padding(
+        page,
+        page.locator("div#codelist-builder-container"),
+        "codelist-builder-empty.png",
+    )
+
+    page.get_by_role("searchbox", name="Enter a search term…").click()
+    page.get_by_role("searchbox", name="Enter a search term…").fill("elbow")
+
+    # Entering a search
+    screenshot_element_with_padding(
+        page,
+        page.locator("div#codelist-builder-container"),
+        "codelist-builder-search.png",
+    )
+
+    page.get_by_role("button", name="Search", exact=True).click()
+
+    # Validate search result
+    page.get_by_role("heading", name='Showing 11 concepts matching "elbow"')
+
+    # Showing a search result
+    screenshot_element_with_padding(
+        page,
+        page.locator("div#codelist-builder-container"),
+        "hierarchy-initial.png",
+    )
+
+    # Expanding the hierarchy
+    codes_to_expand = ["298869002", "298163003", "429554009", "439656005"]
+    for code in codes_to_expand:
+        concept_locator = page.locator(f"[data-code='{code}']")
+        # Iterating through is necessary because for 439656005;
+        # there are two rows to expand.
+        for row in concept_locator.all():
+            row.get_by_role("button", name="⊞").click()
+
+    # Click away to deselect the last expanded code.
+    page.get_by_role("heading").first.click()
+
+    screenshot_element_with_padding(
+        page,
+        page.locator("h5 ~ .builder__container").first.locator(".."),
```

Comment:
thought: Some of the selectors like this are a little bit fragile. However, we don't have any more direct ways of accessing the relevant document element.